documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -141,11 +178,69 @@ include::code_examples/CustomMarshaller.java[]
 include::config_examples/custom_marshaller.xml[]
 ----
 
-
 TIP: It's possible custom marshaller implementations to access a configured white list via the
 link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)
 method, which is called during startup.
 
+== Adding Java Classes to Deserialization White Lists
+{brandname} does not allow deserialization of custom payloads into Java classes
+for security reasons, which applies to JSON, XML, and marshalled `byte[]`
+content.
+
+You must add Java classes to a deserialization white list, either using system
+properties or specifying them in the {brandname} configuration.
+
+.System properties
+
+[source,options="nowrap",subs=attributes+]
+----
+// Specify a comma-separated list of fully qualified class names
+-Dinfinispan.deserialization.whitelist.classes=java.time.Instant,com.myclass.Entity
+
+// Specify a regular expression to match classes
+-Dinfinispan.deserialization.whitelist.regexps=.*
+----
+
+.Declarative
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/deserialization_whitelist.xml[]
+----
+
+== Storing Deserialized Objects
+If you configure {brandname} caches to use the `application/x-java-object`
+MediaType as the storage format (storing POJOs instead of binary content), you
+must put class files for all custom objects on the classpath for {brandname}
+servers.
+
+//dnaro: need to clarify this point
+When utilising a custom marshaller, it is also necessary for the marshaller and it's runtime dependencies
+to be on the server's classpath.  To aid with this step we have created a "bundle" jar for each of the bridge implementations
```

Comment:
create a "bundle" jar for the Kryo and Protostuff marshaller implementations which includes all ..